typing/typecore.ml

```diff
@@ -481,12 +481,13 @@ let type_continuation_pat env expected_ty sp =
 let unify_exp_types loc env ty expected_ty =
   (* Format.eprintf "@[%a@ %a@]@." Printtyp.raw_type_expr exp.exp_type
     Printtyp.raw_type_expr expected_ty; *)
-  try
+  match
     unify env ty expected_ty
   with
-    Unify err ->
+  | Ok () -> ()
+  | Error err ->
       raise(Error(loc, env, Expr_type_clash(err, None, None)))
-  | Tags(l1,l2) ->
+  | exception Tags(l1,l2) ->
```

Comment:
Ah, of course `unify` raises another possible exception, `Tags ...`, which is undocumented in the .mli, has an unclear status (it has its own error handler registered by the compiler... but is also rewrapped as another error in some cases), and remains raised as an exception rather than being part of the `Error` cases.

I think that the choice that you made to leave it as an exception is the correct one, and it's not your fault that it was never documented in the first place, but could you mention it in ctype.mli?